src/vs/workbench/contrib/terminal/common/scripts/shellIntegration-rc.zsh

```diff
@@ -80,18 +80,19 @@ __vsc_escape_value() {
 	builtin emulate -L zsh
 
 	# Process text byte by byte, not by codepoint.
-	builtin local LC_ALL=C str="$1" i byte token out=''
+	builtin local LC_ALL=C str="$1" i byte token out='' val
 
 	for (( i = 0; i < ${#str}; ++i )); do
+	# Escape backslashes, semi-colons specially, then special ASCII chars below space (0x20).
 		byte="${str:$i:1}"
-
-		# Escape backslashes, semi-colons and newlines
-		if [ "$byte" = "\\" ]; then
+		val=$(printf "%d" "'$byte")
+		if (( val < 31 )); then
+			# For control characters, use hex encoding
+			token=$(printf "\\\\x%02x" "'$byte")
+		elif [ "$byte" = "\\" ]; then
```

Comment:
Do we need to do the same for fish?

https://github.com/microsoft/vscode/blob/bd077e517b99f927d4ad60d1f47ff485d986f0ed/src/vs/workbench/contrib/terminal/common/scripts/shellIntegration.fish#L77-L80

Reply:
I think https://github.com/microsoft/vscode/blob/bd077e517b99f927d4ad60d1f47ff485d986f0ed/src/vs/workbench/contrib/terminal/common/scripts/shellIntegration.fish#L95-L101 is what you meant.

I couldn't repro the problem on fish (with starship prompt). 

<img width="1312" alt="Screenshot 2025-03-18 at 3 58 34 PM" src="https://github.com/user-attachments/assets/2222cb4f-3034-4aa9-a423-1999da3f5cfa" />

Perhaps its still worth doing this to be insure better for future though.